### PR TITLE
[12.0][ADD] account_invoice_mass_sending

### PR DESCRIPTION
**Pull request by RobinetDenisAcsone**
Add a new addon which allow to send multiples invoices in jobs.

### CODE REVIEW

**Comment by RobinetDenisAcsone** [comment]
cc @ThomasBinsfeld

File: account_invoice_mass_sending/README.rst [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+Authors
+~~~~~~~
+
+* ACSONE SA/NV
+
+Contributors
+~~~~~~~~~~~~
+
+* Thomas Binsfeld <thomas.binsfeld@acsone.eu>
```
Comment (ThomasBinsfeld) [inline account_invoice_mass_sending/README.rst:71]: @RobinetDenisAcsone Add yourself here

File: account_invoice_mass_sending/README.rst [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+**Table of contents**
+
+.. contents::
+   :local:
+
+Usage
+=====
+
+On the invoice list view, select the invoice to send and click on 'Action > Send & print'.
```
Comment (ThomasBinsfeld) [inline account_invoice_mass_sending/README.rst]: @RobinetDenisAcsone s/invoice/invoices

File: account_invoice_mass_sending/README.rst [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+It will used the existing 'SEND & PRINT' action on the invoice with the default value.
```
Comment (ThomasBinsfeld) [inline account_invoice_mass_sending/README.rst]: @RobinetDenisAcsone s/used/use

File: account_invoice_mass_sending/readme/DESCRIPTION.rst [hunk 4]
```diff
@@ -0,0 +1 @@
+Send invoices in mass mode.
```
Comment (ThomasBinsfeld) [inline account_invoice_mass_sending/readme/DESCRIPTION.rst]: @RobinetDenisAcsone This module adds a mass sending feature on invoices, using asynchronous jobs.

File: account_invoice_mass_sending/README.rst [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
```
Comment (ThomasBinsfeld) [inline account_invoice_mass_sending/README.rst:49]: @RobinetDenisAcsone You can add a configuration section explaining that there are two channels to configure.

File: account_invoice_mass_sending/models/account_invoice.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+from odoo.exceptions import UserError
+
+JOB_QUEUE_CHANNEL = "root.PREPARE_SEND_PRINT_INVOICE"
+SEND_QUEUE_CHANNEL = "root.SEND_PRINT_INVOICE"
+
+
+class AccountInvoice(models.Model):
+    _inherit = "account.invoice"
+
+    sending_in_progress = fields.Boolean(default=False)
```
Comment (ThomasBinsfeld) [inline account_invoice_mass_sending/models/account_invoice.py]: @RobinetDenisAcsone We could add a `help` explaining what this field is about.

File: account_invoice_mass_sending/models/account_invoice.py [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+    @api.multi
+    def mass_send_print(self):
```
Comment (ThomasBinsfeld) [inline account_invoice_mass_sending/models/account_invoice.py:21]:
```suggestion
    def mass_send_print(self):
    """
    This method triggers the asynchronous "Send & Print" for the selected invoices for which there is no asynchronous sending in progress. 
    """
```

**Review by ThomasBinsfeld — COMMENTED** [review]
Mainly nitpicking comments, it looks good.

**Review by ThomasBinsfeld — APPROVED** [review]

**Review by misern2 — APPROVED** [review]
👍

**Comment by legalsylvain** [comment]
Hi @RobinetDenisAcsone. Could you fix conflicts ? 
regards.

**Comment by misern2** [comment]
Hi @RobinetDenisAcsone . Could you fix conflicts ?

Thanks!.

**Pull request merged**